crates/cli/src/codegen/builder.rs

```diff
@@ -44,11 +47,25 @@ impl WitOptions {
     }
 }
 
+/// Strategy for determining the import namespace to use.
+pub(crate) enum ImportNamespace {
+    /// Use javy_quickjs_provider_v2.
+    JavyQuickJsProviderV2,
+    /// Get the import namespace from the provider.
+    FromProvider,
+}
+
+impl Default for ImportNamespace {
+    fn default() -> Self {
+        Self::FromProvider
+    }
+}
+
 /// A code generation builder.
 #[derive(Default)]
 pub(crate) struct CodeGenBuilder {
-    /// The QuickJS provider module version.
-    provider_version: Option<&'static str>,
+    /// The import namespace for dynamically linked modules.
+    import_namespace: Option<ImportNamespace>,
```

Comment:
Is there a particular reason why this can't be `import_namespace: ImportNamespace` and default to `FromProvider`? i.e., drop the `Option`?

Reply:
Funnily enough that was my first implementation. Then I added back the option to bring it closer to how it's currently implemented on `main`. I'll drop the option and set the default to `FromProvider`.

Reply:
Actually, what I'll end up doing is having the provider do something polymorphically instead based on just merging #787.